### PR TITLE
New Pokemon Selector for GUI

### PR DESCRIPTION
**Pull request by cecpk**
## Description
Delete Select2 and add div for selecting Pokemon with searchform

Searchoption:
**ID
Name
Type
Gen**

Allow combined search with "," and "+" or "-"

f.E. looking for all dragons of gen1: dragon,-gen2,-gen3
or
looking for all gen1 pokemons without Nr 1,2,3: gen1,-1,-2,-3

Ready for:
- Hide Pokemon
- Notify Pokemon

## Motivation and Context
Mobile usage for select2 not working. List like PMSF

## Screenshots (if appropriate):
screenshot 2018-04-08 um 15 32 50

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by darkelement1987** [comment]
Tested this. Works pretty good. Unhiding takes a few secs and hiding from map doesn't trigger the button in your menu:

image

I pressed 'Hide' but the Venusaur button was not activated.

**Comment by tomballgithub** [comment]
Seems to me like most users would rather have the pokemon name than the # in the selection screen.  Knowing the pokedex #'s isn't something I would expect my users to know.

**Comment by darkelement1987** [comment]
I dont think that fits properly but its worth a try?

**Comment by The-Real-Ketchum-Dev** [comment]
I would love the names instead but i have to agree its worth the try and dont think theyll fit at least not all of them unless they were small hard to see names lol

**Comment by bcrucialhd** [comment]
schiggy tho!

**Comment by darkelement1987** [comment]
Bug: When typing something in either searchboxes, it will apply to both:

I entered: 'Corsola @ Hide'
image

Then i went to notifications but entered nothing:

image

**Comment by cecpk** [comment]
travis sucks ... dont found any of this errors ....

**Comment by darkelement1987** [comment]
@rlodev67 i don't see it either :( could you maybe help?

**Comment by darkelement1987** [comment]
Few suggestions (your choice to code this or not, i'm already satisfied with the code as it is :D )

**Suggestions:**
- Add 'comma-seperated'-searching
- Make searchfilter find everything starting with (for example) 15 when you enter `15`. So that includes 150, 151, 152, 153 etcetera. At this moment `15` will only show Pokemon number 15.
- Searchfilter by generation
- Option to load/export "hide-lists" and support in custom.js. (inspired by the 'Favorite locations'-PR from Stock RM)

**Comment by darkelement1987** [comment]
https://github.com/cecpk/RocketMap/pull/4

Solved Travis for your new commits.

File: static/js/map.js [hunk 1]
```diff
@@ -3153,6 +3326,27 @@ $(function () {
         }
     }
 
+    Array.prototype.diff = function (a) {
```
Comment (darkelement1987) [inline static/js/map.js]: Stuff @ Line 3329-3333 gives errors @ Travis CI:

> error Array prototype is read only, properties should not be added no-extend-native

**Comment by cecpk** [comment]
...

**Comment by darkelement1987** [comment]
Why

**Pull request closed without merging**